src/main.test.ts

```diff
@@ -0,0 +1,136 @@
+import * as fs from 'fs'
+import { join } from 'path'
+import * as path from 'path'
+import * as process from 'process'
+
+import * as Diff from 'diff'
+import { test } from 'uvu'
+
+import { Input } from './Input'
+import * as lsif from './lsif'
+import { index as lsifIndex } from './main'
+import { Range } from './Range'
+
+const lsif_typed = lsif.lib.codeintel.lsif_typed
+
+function isUpdateSnapshot(): boolean {
+  return process.argv.includes('--update-snapshots')
+}
+
+const inputDirectory = join(process.cwd(), 'snapshots', 'input')
+const outputDirectory = join(process.cwd(), 'snapshots', 'output')
+
+const snapshotDirectories = fs.readdirSync(inputDirectory)
+const isUpdate = isUpdateSnapshot()
+if (isUpdate && fs.existsSync(outputDirectory)) {
+  fs.rmSync(outputDirectory, { recursive: true })
+}
+for (const snapshotDirectory of snapshotDirectories) {
+  test(snapshotDirectory, () => {
+    const index = new lsif.lib.codeintel.lsif_typed.Index()
+    const projectRoot = join(inputDirectory, snapshotDirectory)
+    lsifIndex({
+      projectRoot,
+      project: projectRoot,
+      writeIndex: partialIndex => {
+        if (partialIndex.metadata) {
+          index.metadata = partialIndex.metadata
+        }
+        for (const document of partialIndex.documents) {
+          index.documents.push(document)
+        }
+      },
+    })
+    // eslint-disable-next-line no-sync
+    fs.writeFileSync(
+      path.join(projectRoot, 'dump.lsif-typed'),
```

Comment:
This generates `dump.lsif-typed` files under `snapshots/input` (instead of in the `snapshots/output/` directory) when you run tests, which feels a little odd. Intentional or accidental?